### PR TITLE
[wasm][debugger] Add support for Nullable<T> 

### LINKED ISSUE #20310: [wasm][debugger] `Nullable<T>`'s value not shown correctly
Created based on https://github.com/mono/mono/issues/19942#issuecomment-671557546

### PR DESCRIPTION
**Pull request by radical**
Return the value, or null.

Fixes mono/mono#20310

### CODE REVIEW

**Comment by radical** [comment]
This is based on https://github.com/dotnet/runtime/pull/41480 , so it includes changes from that.
The additional commit for this PR: https://github.com/dotnet/runtime/pull/41559/commits/dd47793bf6fa190f80bef7496dad522149d62013

**Comment by radical** [comment]
Rebased on master. Ready for review

**Review by lewing — COMMENTED** [review]
Why split out the two functions for has_value and _value that are both only used in one place?

File: src/mono/mono/mini/mini-wasm-debugger.c [hunk 1]
```diff
@@ -900,6 +936,23 @@ describe_value(MonoType * type, gpointer addr, int gpflags)
...
+
+				gpointer has_value_addr = nullable_get_has_value_field_addr (addr, klass);
+				gboolean has_value = *(guint8*)has_value_addr;
+				if (!has_value) {
+					char* class_name = mono_type_full_name (type);
+					mono_wasm_add_obj_var (class_name, NULL, 0);
+					g_free (class_name);
+					break;
+				} else {
+					gpointer value_addr = nullable_get_value_field_addr(addr, klass);
```
Comment (lewing) [inline src/mono/mono/mini/mini-wasm-debugger.c:951]: inconsistent formatting

**Comment by radical** [comment]
> Why split out the two functions for has_value and _value that are both only used in one place?

I had copied this from `mono/metadata/object.c`. I can merge them.

File: src/mono/wasm/debugger/tests/debugger-nullable-test.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+#nullable enable
+
+namespace DebuggerTests
+{
+    public class NullableTests
+    {
+        public static void TestNullableLocal ()
```
Comment (lewing) [inline src/mono/wasm/debugger/tests/debugger-nullable-test.cs:11]: more formatting nits in this file

File: src/mono/mono/mini/mini-wasm-debugger.c [hunk 3]
```diff
@@ -829,40 +829,24 @@ read_enum_value (const char *mem, int type)
 	return 0;
 }
 
-static MonoClassField*
-nullable_class_get_value_field (MonoClass *klass)
+static gpointer
+nullable_get_has_value_field_addr (guint8 *nullable, MonoClass *klass)
```
Comment (lewing) [inline src/mono/mono/mini/mini-wasm-debugger.c:833]: could this just be `static gboolean nullable_get_has_value` ?

**Review by lewing — APPROVED** [review]
Nice!

**Pull request merged**